File: dotnet/Endpoints/Create.cs
using Microsoft.AspNetCore.Http.HttpResults;
using Microsoft.AspNetCore.Mvc;
using Todo.Core.Abstract.Services;
using Todo.Endpoints.Mappers;
using Todo.Endpoints.Models;

namespace Todo.Endpoints;

internal class Create
{
    public static async Task<Ok<TodoDto>> Execute(
        [FromBody] MutateTodoDto dto,
        [FromServices] ITodoService todoService
    )
    {
        var saved = await todoService.Create(dto.ToEntity());

        return TypedResults.Ok(saved.ToDto());
    }
}

File: nodejs/src/endpoints/todos/create.ts
import { ITodoService } from "../../core/abstract/services/todo-service";
import { Request, Response } from "express";
import { ToTodo } from "../mappers/create-todo-dto.mappers";
import { ToTodoDto } from "../mappers/todo-dto.mappers";

export const CreateTodoEndpoint = (todoService: ITodoService) => {
  return async (req: Request, res: Response) => {
    let todo = ToTodo(req.body);

    todo = await todoService.create(todo);

    res.status(200).json(ToTodoDto(todo));
  };
};
